crates/iota-network/src/discovery/mod.rs

```diff
@@ -212,18 +212,36 @@ impl DiscoveryEventLoop {
         }
     }
 
-    // TODO: we don't boot out old committee member yets, however we may want to do
-    // this in the future along with other network management work.
-    /// Handles a [`TrustedPeerChangeEvent`] by adding the new trusted peers to
-    /// known peers list.
+    /// Handles a [`TrustedPeerChangeEvent`] by updating the known peers with
+    /// the latest trusted new peers without deleting the allowlisted peers
     fn handle_trusted_peer_change_event(
         &mut self,
         trusted_peer_change_event: TrustedPeerChangeEvent,
```

Comment:
Can't we figure out when the `TrustedPeerChangeEvent` gets created, which peers should be added and which should be removed? Ofc the "peers_to_remove" list should not remove peers that are part of the "allowlisted_peers". This is the initial list of trusted peers from the config, right?

Reply:
We figured it! The config contains allowlisted and seed peers; we implemented the event handling logic so that they can only be updated (if metadata changed) but not removed.